### DISCUSSION TITLE
nvim-autopair treesitter integration

### DISCUSSION THREAD

**Discussion by Trid-collab** [nvim-treesitter/nvim-treesitter#4046] [General]
I have using nvim-autopairs in my config since early this year. The setup included the following line in my treesitter config apart from the config in the autopairs setup. `autopairs = {enable = true}`

While revisiting my nvim config files for a rewrite I find no reference of the above requirement for treesitter config mentioned in the autopair or the the treesitter page. Is it not required now

**Comment by theHamsta** [comment]
This was from a different module. Possibly this plugin: https://github.com/windwp/nvim-autopairs/blob/3909fc7912f4349ee8b8891056ad807abc7221d3/tests/treesitter_spec.lua#L18 Maybe they used to use a nvim-treesitter module.

Other plugins can register modules. nvim-treesitter will then assist when they get attached/detached from the buffer.
